platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,168 @@
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile,
+                        // Use the latest value for the same key
+                        (pid1, pid2) -> pid2));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        try {
+            switch (configurationEvent.getType()) {
+            case CM_UPDATED:
+                LOGGER.debug("Handling update");
+                handleUpdate(new FelixConfig(() -> getConfiguration(configurationEvent)));
+                break;
+            case CM_DELETED:
+                LOGGER.debug("Handling delete");
+                handleDelete(configurationEvent.getPid());
+                break;
+            default:
+                LOGGER.debug("Unknown configuration event type, taking no action");
+            }
+        } catch (IOException e) {
+            LOGGER.debug("Error reading or modifying configurations on disk: {}", e);
+        } catch (IllegalArgumentException e) {
+            LOGGER.debug("Encountered bad arguments: {}", e);
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param config the {@link Configuration} wrapped in a {@link FelixConfig} that arrived as an event.
+     * @throws IOException if an issue occurs persisting to disk.
+     */
+    private void handleUpdate(FelixConfig config) throws IOException {
+        String pid = config.getPid();
+        File felixFileNew = config.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        boolean filePropChanged = config.filePropChanged(felixFileOriginal);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
+            LOGGER.debug("Will not track pid {}", pid);
+            return;
+        }
+
+        if (filePropChanged) {
+            if (felixFileOriginal != null) {
+                LOGGER.debug("Felix filename prop changed, should delete file for pid {}", pid);
+                Files.delete(felixFileOriginal.toPath());
+                if (felixFileNew == null) {
+                    LOGGER.debug("No longer tracking file for pid {}", pid);
+                    pidFileMap.remove(pid);
+                    return;
+                }
+            }
+            LOGGER.debug("Tracking pid {}", pid);
+            pidFileMap.put(pid, felixFileNew);
+        }
+    }
+
+    /**
+     * Remove config files from etc when a delete occurs.
+     *
+     * @param pid the pid of the configuration that was deleted.
+     * @throws IOException if an issue occurs persisting to disk.
+     */
+    private void handleDelete(String pid) throws IOException {
+        File felixFileOriginal = pidFileMap.remove(pid);
+        if (felixFileOriginal != null && felixFileOriginal.isFile()) {
+            LOGGER.debug("Deleting file because config was deleted for pid {}", pid);
+            Files.delete(felixFileOriginal.toPath());
```

Comment:
I don't care where you put the logic, but before deleting or modifying any files on the file system, you need to check that those files are within the expected directory structure and you also need to add audits.